### PR TITLE
Deprecate verbosity macros (not compatible w/ juliac-compiled programs)

### PR DESCRIPTION
**Pull request by quinnj**
Add convenience group keyword to withlevel to do early filtering on groups

Export withlevel

### CODE REVIEW

File: src/verbosity.jl [hunk 1]
```diff
@@ -7,7 +7,15 @@ function restore_callsite_source_position!(expr, src)
...
-    return expr
+    ex = quote
+        LoggingExtras.deprecate_verbosity($(Meta.quot(m)))
+        $expr
+    end
+    return ex
+end
+
+function deprecate_verbosity(m)
+    Base.depwarn("Verbosity logging macros are deprecated as they are not compatible with juliac-compiled programs", m; force=true)
```
Comment (oxinabox) [inline src/verbosity.jl:18]:
```suggestion
    Base.depwarn("Verbosity logging macros are deprecated as they are not compatible with juliac-compiled programs", m)
```

I see no reason to `force`.

File: src/verbosity.jl [hunk 2]
```diff
@@ -34,56 +42,57 @@ end
...
 log messages will first be checked that they meet the `level`
 log level before being passed on to be logged.
 
-For convenience, a `verbosity` keyword argument can be passed which also
-filters the "verbose logging" messages; see [`@debugv`](@ref), [`@infov`](@ref),
-[`@warnv`](@ref), [`@errorv`](@ref), and [`@logmsgv`](@ref).
+For convenience, a `group` keyword argument can be passed which also
+filters logging messages on the "group". By default, the group is the
+file name of the log macro call site, but can be overridden by passing
+the `_group` keyword argument to the logging macros.
```
Comment (oxinabox) [inline src/verbosity.jl:95]: I am not conviced thsi is better than just using

```
with_logger(EarlyFilteredLogger(x->x.group==:Foo) do
...
end
```

Maybe we should instead be adding a 
```
filter_group_logger(group) = EarlyFilteredLogger(x->x.group==group)
```
if it really is too long

Comment (quinnj) [reply src/verbosity.jl:95]: It's really the convenience of having it bundled _with_ the _withlevel_ function. It's so onerous remembering the incantation to get the logger, set it to the new one, and then on top remember the LoggingExtras machinery to early filter. Bundling that all up into `withlevel` as the "one thing" I need to remember as a package dev is what makes this work really well.

File: src/verbosity.jl (hunk 1 above)
Comment (quinnj) [reply src/verbosity.jl]: Ah yes, I was just testing that the depwarn actually worked. Good catch.

**Pull request merged**